src/flinkStatementResultsManager.ts

```diff
@@ -227,11 +230,7 @@ export class FlinkStatementResultsManager {
         // Convert Map to array of objects
         const res = Array.from(this._results().values()).map((row: Map<string, any>) => {
           // Convert Map to plain object
-          const obj: Record<string, any> = {};
-          row.forEach((value: any, key: string) => {
-            obj[key] = value;
-          });
-          return obj;
+          return this.mapToObject(row);
```

Comment:
Does [`Object.fromEntries(row)`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/fromEntries) help here? Hopefully we shouldn't need a dedicated conversion method from `Map` to `Record` unless there's some special converting that needs to happen

Reply:
> 2 passing (2s)

yes it does help, thanks!